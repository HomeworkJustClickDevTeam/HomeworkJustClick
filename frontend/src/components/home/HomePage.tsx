import React, {useContext, useEffect, useState} from "react"
import GroupListElement from "../group/GroupListElement"
import {Link, useNavigate} from "react-router-dom"
import {groupFilter} from "../group/filter/GroupFilter"
import Loading from "../animations/Loading"
import DispatchContext from "../../contexts/DispatchContext"
import {FaCaretDown} from 'react-icons/fa';
import {GroupInterface} from "../../types/GroupInterface";
import {ActionType} from "../../types/ActionType";
import {getUser} from "../../services/otherServices";

function HomePage() {
  const [groups, setGroups] = useState<GroupInterface[] | undefined>(undefined)
  const [isLoading, setIsLoading] = useState<boolean>()
  const [isOpen, setIsOpen] = useState(false)
  const [btnName, setBtnName] = useState('Wszystkie grupy')
  const navigate = useNavigate()
  const userState = getUser()
  const globalDispatch = useContext(DispatchContext)
  let teacherUserGroups = () => {}
  let studentsUserGroups = () => {}
  let allUserGroups = () => {}

  if(userState !== undefined){
    ({teacherUserGroups, studentsUserGroups, allUserGroups } = groupFilter({
      setGroups,
      setIsLoading,
      userId: userState.id.toString()
    }))
  }
  useEffect(() => {
    const action: ActionType = {
      type: "homePageIn",
    }
    globalDispatch?.dispatch(action)
    allUserGroups()
  }, [])

  if (groups === undefined || groups === null) {
    return <Loading />
  }
  if(userState === undefined){
    return <>User Not Logged in</>
  }


  return (
    <div className='pl-6 pr-8'>
        <div className='flex inline-block mb-4 mt-4 pl-5'>
            <p className='pr-8 text-2xl'>Moje grupy</p>
          <Link to="/create/group" className=''>
            <button type="button" className='text-black text-base underline items-center pt-2'>Stwórz grupę +</button>
          </Link>
          <div className='absolute inline-block text-left right-0 mr-16'>
            <div className='flex inline'>
              <span className='mr-2'>Wyświetlane grupy: </span>
              <button className='flex inline underline' onClick={() => setIsOpen(el => !el)}>{btnName}<FaCaretDown className=' ml-1 h-full w-[9px] items-center'/>
              </button>
            </div>

            {isOpen && (
                <div className=' absolute left-20 z-10 w-56 mt-4 origin-top-right bg-white border border-hover_gray rounded-md shadow-lg'>
                  <div onClick={() => {studentsUserGroups(); setBtnName('Grupy uczniowskie')}} className="block px-4 py-2 text-sm text-black rounded-lg hover:bg-lilly-bg hover:text-black">
                    Grupy uczniowskie użytkownika
                  </div>
                  <div onClick={() => {teacherUserGroups(); setBtnName('Grupy nauczycielskie')}} className="block px-4 py-2 text-sm text-black rounded-lg hover:bg-lilly-bg hover:text-black">
                    {" "}
                    Grupy nauczycielskie użytkownika
                  </div>
                  <div onClick={() => {allUserGroups(); setBtnName('Wszystkie grupy')}} className="block px-4 py-2 text-sm text-black rounded-lg hover:bg-lilly-bg hover:text-black">Wszystkie grupy użytkownika</div>
                </div>
            )}

          </div>
        </div>
      {isLoading ? (
        <Loading />
      ) : (
        <div className='relative flex flex-wrap gap-x-5 gap-y-2 bg-light_gray px-8 py-6 bg-lilly-bg rounded-md'>
          {groups?.map((group) =>
              groups?.length > 0 ? <GroupListElement group={group} key={group.id} /> : <div>Dodaj nową grupę!</div>
          )}
        </div>
      )}

    </div>
  )
}
export default HomePage
