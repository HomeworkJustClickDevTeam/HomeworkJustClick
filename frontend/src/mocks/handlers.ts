import { http, HttpResponse, delay } from "msw";
import { be } from "date-fns/locale";


export const homePageHandlers = [
  http.get('http://localhost:8080/api/groups/byTeacher/:id', async ({params}) => {
    await delay(100)
    if(params.id !== "undefined" && params.id !== "null" && params.id === '0'){
      return HttpResponse.json(null, {status: 404})
    }
    return HttpResponse.json([
        {
          id: 0,
          name: "Example Group 1",
          description: "Example desc",
          color: 0,
          archived: false
        },
        {
          id: 2,
          name: "Example Group 2",
          description: "Example desc 2",
          color: 1,
          archived: false
        }
      ],
      {
        status: 200
      })
  }),
  http.get('http://localhost:8080/api/groups/byStudent/:id', async ({params}) => {
    await delay(100)
    return HttpResponse.json(null, {status: 404})
  }),
  http.get('http://localhost:8080/api/groups/byUser/:id', async ({params}) => {
    const {id} = params
    await delay(100)
    if(params.id !== "undefined" && params.id !== "null" && params.id === '0'){
      return HttpResponse.json(null, {status: 404})
    }
    return HttpResponse.json( [
          {
            id: 0,
            name: "Example Group 1",
            description: "Example desc",
            color: 0,
            archived: false
          },
          {
            id: 2,
            name: "Example Group 2",
            description: "Example desc 2",
            color: 1,
            archived: false
          }
        ],
      {
        status: 200
      })
  }),
  http.get('http://localhost:8080/api/groups/userCheckWithRole/:userId/:groupId', async ({params}) => {
    const {userId, groupId} = params
    await delay(100)
    if(userId === '3'){
      return new Response("Teacher")
    }
    else
      return new Response("Student")
  })
]

export const loginPageHandler = [
  http.post('http://localhost:8080/api/auth/authenticate', async () => {

    await delay(100)
    return HttpResponse.json({
        token: "eyJhbGc3ZGEuY29tIiwiaWF0IjoxNzAwOTkwNjkyLCJleHAiOjE3MDA5OTQyOTJ9.B22HfZ1nIWfmHB66lt0vppGE86eERfTM-f61GG--v0LB-7dulmb4H11Q3niO4NYSJncEE5JzRhoBHhdraIP3og",
        id: 1,
        role: "USER",
        message: "ok",
        color: 0,
        name: "Jan",
        lastname: "Kowalski",
        index: 123123
      }, {
      status: 200
    })
  })
]