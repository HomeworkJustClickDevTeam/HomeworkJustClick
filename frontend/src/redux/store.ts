import {combineReducers, configureStore, PreloadedState} from "@reduxjs/toolkit"
import userStateReducer from "./userStateSlice"
import roleReducer from "./roleSlice"
import groupReducer from "./groupSlice"
import isLoadingReducer from "./isLoadingSlice"
import homePageInReducer from "./homePageInSlice"



const rootReducer = combineReducers({
  userState: userStateReducer,
  role: roleReducer,
  group: groupReducer,
  isLoading: isLoadingReducer,
  homePageIn: homePageInReducer
})
export const setupStore = (preloadedState?: PreloadedState<RootState>) =>{
  return configureStore({
    reducer: rootReducer,
    preloadedState
  })
}


export type RootState = ReturnType<typeof rootReducer>
export type AppStore = ReturnType<typeof setupStore>
export type AppDispatch = AppStore['dispatch']