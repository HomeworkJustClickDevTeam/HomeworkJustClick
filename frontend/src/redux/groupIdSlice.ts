import {createSlice, PayloadAction} from "@reduxjs/toolkit";
import {RootState} from "./store";

type GroupIdSliceType = number | null
const initialState = null as GroupIdSliceType

export const groupIdSlice = createSlice({
    name: 'groupId',
    initialState,
    reducers: {
        setGroupId: (state, action: PayloadAction<GroupIdSliceType>) => action.payload
    }
})

export const {setGroupId} = groupIdSlice.actions
export const selectGroupId = (state: RootState) => state.groupId
export default groupIdSlice.reducer