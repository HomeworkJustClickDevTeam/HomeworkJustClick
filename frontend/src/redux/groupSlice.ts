import { createSlice, PayloadAction } from "@reduxjs/toolkit"
import { GroupInterface } from "../types/GroupInterface"
import { RootState } from "./store"

type GroupSliceType = GroupInterface | null
const initialState = null as GroupSliceType
export const groupSlice = createSlice({
  name:'group',
  initialState,
  reducers:{
    setGroup:(state, action:PayloadAction<GroupSliceType>) => action.payload
  }
})

export const {setGroup} = groupSlice.actions
export const selectGroup = (state:RootState) => state.group
export default groupSlice.reducer