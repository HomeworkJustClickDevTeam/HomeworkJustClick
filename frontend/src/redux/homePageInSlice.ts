import { createSlice, PayloadAction } from "@reduxjs/toolkit"
import { RootState } from "./store"

const initialState: boolean = true
export const homePageInSlice = createSlice({
  name: 'homePageIn',
  initialState,
  reducers:{
    setHomePageIn:(state, action:PayloadAction<boolean>) => action.payload
  }
})

export const {setHomePageIn} = homePageInSlice.actions
export const selectHomePageIn = (state:RootState) => state.homePageIn
export default homePageInSlice.reducer