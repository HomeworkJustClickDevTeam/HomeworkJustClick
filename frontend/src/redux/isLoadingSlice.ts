import { createDraftSafeSelector, createSlice } from "@reduxjs/toolkit"
import { AppDispatch, RootState } from "./store"

const initialState: boolean = false

export const isLoadingSlice = createSlice({
  name: 'isLoading',
  initialState,
  reducers:{
    setIsLoading: (_, action) => action.payload
  }
})

export const {setIsLoading} = isLoadingSlice.actions

export const setIsLoadingInReducer = (newState:boolean) => {
  return (dispatch:AppDispatch) => {
    dispatch(setIsLoading(newState))
  }
}
export const selectIsLoading = (state:RootState):boolean => state.isLoading //Use when you want to obtain the state of isLoading
export default isLoadingSlice.reducer