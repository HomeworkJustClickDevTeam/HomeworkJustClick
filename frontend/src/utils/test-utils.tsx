import {configureStore, PreloadedState} from "@reduxjs/toolkit";
import userStateReducer from "../redux/userStateSlice";
import roleReducer from "../redux/roleSlice";
import groupReducer from "../redux/groupSlice";
import isLoadingReducer from "../redux/isLoadingSlice";
import homePageInReducer from "../redux/homePageInSlice";
import {Provider} from "react-redux";
import {render, RenderOptions} from "@testing-library/react";
import {RootState, setupStore} from "../redux/store";
import {ToolkitStore} from "@reduxjs/toolkit/dist/configureStore";
import {PropsWithChildren} from "react";

interface ExtendedRenderOptions extends Omit<RenderOptions, 'queries'> {
  preloadedState?: PreloadedState<RootState>
  store?: ToolkitStore
}
export const renderWithProviders = (
    ui: React.ReactElement,
    {
      preloadedState = {},
      store = setupStore(preloadedState),
      ...renderOptions
    }: ExtendedRenderOptions={})=> {

  const Wrapper = ({children}:PropsWithChildren<{}>):JSX.Element => {
    return <Provider store={store}>{children}</Provider>
  }

  return {store, ...render(ui, {wrapper: Wrapper, ...renderOptions})}
}