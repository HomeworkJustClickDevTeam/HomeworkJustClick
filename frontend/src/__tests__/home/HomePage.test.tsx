import { setupServer } from "msw/node";
import { homePageHandlers } from "../../mocks/handlers";
import '@testing-library/jest-dom';
import { BrowserRouter } from "react-router-dom";
import HomePage from "../../components/home/HomePage";
import renderer from "react-test-renderer";
import { Provider } from "react-redux";
import { setupStore } from "../../redux/store";
import React from "react";
import { tr } from "date-fns/locale";
import { renderWithProviders } from "../../utils/test-utils";
import userEvent from "@testing-library/user-event";
import { screen } from "@testing-library/react";
import { use } from "msw/lib/core/utils/internal/requestHandlerUtils";

const server = setupServer(...homePageHandlers)
beforeAll(() => server.listen())
beforeEach(()=> {
  jest.clearAllMocks()
})
afterEach(()=> server.resetHandlers())
afterAll(() => server.close())


it("checks if home page renders correctly with sample groups",  () => {
  const renderedHomePage = renderer
    .create(
      <Provider store={setupStore({userState: {
          token: "eyJhbGciOiJIUzUxMiJ9.eyJzdWIiOiJhZGFtX25vd2FrQGdtYWlsLmNvbSIsImlhdCI6MTcwMTA2NjIyOCwiZXhwIjoxNzAxMDY5ODI4fQ.gIqB5eSqKchKtFYs17GskZALXXtpVpSFfnJehuSfRNt3EVYipbT_UNFH09NaGUvGtAK-Mq4SukhrZoJW_F8eVw",
          id: 3,
          role: "USER",
          color: 2,
          name: "Adam",
          lastname: "Nowak",
          index: 222222
        }})}>
        <BrowserRouter>
          <HomePage></HomePage>
        </BrowserRouter>
      </Provider>
    )
  expect(renderedHomePage).toMatchSnapshot()
})

it("checks if home page group sorting works",  async () => {
  const user = userEvent.setup()
  const ui =
    <BrowserRouter>
      <HomePage></HomePage>
    </BrowserRouter>
  const {container, rerender} = renderWithProviders(
          ui,
    {preloadedState:{userState: {
          token: "eyJhbGciOiJIUzUxMiJ9.eyJzdWIiOiJhZGFtX25vd2FrQGdtYWlsLmNvbSIsImlhdCI6MTcwMTA2NjIyOCwiZXhwIjoxNzAxMDY5ODI4fQ.gIqB5eSqKchKtFYs17GskZALXXtpVpSFfnJehuSfRNt3EVYipbT_UNFH09NaGUvGtAK-Mq4SukhrZoJW_F8eVw",
          id: 3,
          role: "USER",
          color: 2,
          name: "Adam",
          lastname: "Nowak",
          index: 222222
        }}})
  expect(container).toMatchSnapshot()
  const sortGroupsButton = await screen.findByRole('button', {name:/wszystkie grupy/i})
  expect(sortGroupsButton).toBeInTheDocument()
  await user.click(sortGroupsButton)
  rerender(ui)
  const studentGroups = await screen.findByText(/Grupy uczniowskie u≈ºytkownika/i)
  expect(studentGroups).toBeInTheDocument()
  expect(container).toMatchSnapshot()
  await user.click(studentGroups)
  rerender(ui)
  expect(container).toMatchSnapshot()
})



