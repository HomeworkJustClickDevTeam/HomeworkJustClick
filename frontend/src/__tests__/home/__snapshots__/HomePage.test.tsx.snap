// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`checks if home page group sorting works 1`] = `
<DocumentFragment>
  <div
    class="pl-6 pr-8"
  >
    <div
      class="flex inline-block mb-4 mt-4 pl-5"
    >
      <p
        class="pr-8 text-2xl"
      >
        Moje grupy
      </p>
      <a
        class=""
        href="/create/group"
      >
        <button
          class="text-black text-base underline items-center pt-2"
          type="button"
        >
          Stwórz grupę +
        </button>
      </a>
      <div
        class="absolute inline-block text-left right-0 mr-16"
      >
        <div
          class="flex inline"
        >
          <span
            class="mr-2"
          >
            Wyświetlane grupy: 
          </span>
          <button
            class="flex inline underline"
          >
            Wszystkie grupy
            <svg
              class=" ml-1 h-full w-[9px] items-center"
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 320 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M31.3 192h257.3c17.8 0 26.7 21.5 14.1 34.1L174.1 354.8c-7.8 7.8-20.5 7.8-28.3 0L17.2 226.1C4.6 213.5 13.5 192 31.3 192z"
              />
            </svg>
          </button>
        </div>
      </div>
    </div>
    <div
      class="relative flex flex-wrap gap-x-5 gap-y-2 bg-light_gray px-8 py-6 bg-lilly-bg rounded-md"
    >
      <div
        class="relative w-64 bg-[#59007f]  min-h-fit h-40 overflow-hidden rounded-md"
      >
        <a
          class="w-full"
          href="/group/0"
        >
          <h1
            class="flex items-center justify-center text-center pb-2 h-[50%] pt-3 text-white font-semibold text-lg"
          >
            Example Group 1
          </h1>
           
        </a>
        <p
          class="px-2 pl-2 flex items-center text-base h-[50%] text-ellipsis bg-white bottom-0 text-black"
        >
          Example desc
        </p>
      </div>
      <div
        class="relative w-64 bg-[#006400]  min-h-fit h-40 overflow-hidden rounded-md"
      >
        <a
          class="w-full"
          href="/group/2"
        >
          <h1
            class="flex items-center justify-center text-center pb-2 h-[50%] pt-3 text-white font-semibold text-lg"
          >
            Example Group 2
          </h1>
           
        </a>
        <p
          class="px-2 pl-2 flex items-center text-base h-[50%] text-ellipsis bg-white bottom-0 text-black"
        >
          Example desc 2
        </p>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`checks if home page group sorting works 2`] = `
<DocumentFragment>
  <div
    class="pl-6 pr-8"
  >
    <div
      class="flex inline-block mb-4 mt-4 pl-5"
    >
      <p
        class="pr-8 text-2xl"
      >
        Moje grupy
      </p>
      <a
        class=""
        href="/create/group"
      >
        <button
          class="text-black text-base underline items-center pt-2"
          type="button"
        >
          Stwórz grupę +
        </button>
      </a>
      <div
        class="absolute inline-block text-left right-0 mr-16"
      >
        <div
          class="flex inline"
        >
          <span
            class="mr-2"
          >
            Wyświetlane grupy: 
          </span>
          <button
            class="flex inline underline"
          >
            Wszystkie grupy
            <svg
              class=" ml-1 h-full w-[9px] items-center"
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 320 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M31.3 192h257.3c17.8 0 26.7 21.5 14.1 34.1L174.1 354.8c-7.8 7.8-20.5 7.8-28.3 0L17.2 226.1C4.6 213.5 13.5 192 31.3 192z"
              />
            </svg>
          </button>
        </div>
        <div
          class=" absolute left-20 z-10 w-56 mt-4 origin-top-right bg-white border border-hover_gray rounded-md shadow-lg"
        >
          <div
            class="block px-4 py-2 text-sm text-black rounded-lg hover:bg-lilly-bg hover:text-black"
          >
            Grupy uczniowskie użytkownika
          </div>
          <div
            class="block px-4 py-2 text-sm text-black rounded-lg hover:bg-lilly-bg hover:text-black"
          >
             Grupy nauczycielskie użytkownika
          </div>
          <div
            class="block px-4 py-2 text-sm text-black rounded-lg hover:bg-lilly-bg hover:text-black"
          >
            Wszystkie grupy użytkownika
          </div>
        </div>
      </div>
    </div>
    <div
      class="relative flex flex-wrap gap-x-5 gap-y-2 bg-light_gray px-8 py-6 bg-lilly-bg rounded-md"
    >
      <div
        class="relative w-64 bg-[#59007f]  min-h-fit h-40 overflow-hidden rounded-md"
      >
        <a
          class="w-full"
          href="/group/0"
        >
          <h1
            class="flex items-center justify-center text-center pb-2 h-[50%] pt-3 text-white font-semibold text-lg"
          >
            Example Group 1
          </h1>
           
        </a>
        <p
          class="px-2 pl-2 flex items-center text-base h-[50%] text-ellipsis bg-white bottom-0 text-black"
        >
          Example desc
        </p>
      </div>
      <div
        class="relative w-64 bg-[#006400]  min-h-fit h-40 overflow-hidden rounded-md"
      >
        <a
          class="w-full"
          href="/group/2"
        >
          <h1
            class="flex items-center justify-center text-center pb-2 h-[50%] pt-3 text-white font-semibold text-lg"
          >
            Example Group 2
          </h1>
           
        </a>
        <p
          class="px-2 pl-2 flex items-center text-base h-[50%] text-ellipsis bg-white bottom-0 text-black"
        >
          Example desc 2
        </p>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`checks if home page group sorting works 3`] = `
<DocumentFragment>
  <div
    class="pl-6 pr-8"
  >
    <div
      class="flex inline-block mb-4 mt-4 pl-5"
    >
      <p
        class="pr-8 text-2xl"
      >
        Moje grupy
      </p>
      <a
        class=""
        href="/create/group"
      >
        <button
          class="text-black text-base underline items-center pt-2"
          type="button"
        >
          Stwórz grupę +
        </button>
      </a>
      <div
        class="absolute inline-block text-left right-0 mr-16"
      >
        <div
          class="flex inline"
        >
          <span
            class="mr-2"
          >
            Wyświetlane grupy: 
          </span>
          <button
            class="flex inline underline"
          >
            Grupy uczniowskie
            <svg
              class=" ml-1 h-full w-[9px] items-center"
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 320 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M31.3 192h257.3c17.8 0 26.7 21.5 14.1 34.1L174.1 354.8c-7.8 7.8-20.5 7.8-28.3 0L17.2 226.1C4.6 213.5 13.5 192 31.3 192z"
              />
            </svg>
          </button>
        </div>
        <div
          class=" absolute left-20 z-10 w-56 mt-4 origin-top-right bg-white border border-hover_gray rounded-md shadow-lg"
        >
          <div
            class="block px-4 py-2 text-sm text-black rounded-lg hover:bg-lilly-bg hover:text-black"
          >
            Grupy uczniowskie użytkownika
          </div>
          <div
            class="block px-4 py-2 text-sm text-black rounded-lg hover:bg-lilly-bg hover:text-black"
          >
             Grupy nauczycielskie użytkownika
          </div>
          <div
            class="block px-4 py-2 text-sm text-black rounded-lg hover:bg-lilly-bg hover:text-black"
          >
            Wszystkie grupy użytkownika
          </div>
        </div>
      </div>
    </div>
    <div
      class="relative flex flex-wrap gap-x-5 gap-y-2 bg-light_gray px-8 py-6 bg-lilly-bg rounded-md"
    />
  </div>
</DocumentFragment>
`;

exports[`checks if home page renders correctly with sample groups 1`] = `
<DocumentFragment>
  <div
    class="pl-6 pr-8"
  >
    <div
      class="flex inline-block mb-4 mt-4 pl-5"
    >
      <p
        class="pr-8 text-2xl"
      >
        Moje grupy
      </p>
      <a
        class=""
        href="/create/group"
      >
        <button
          class="text-black text-base underline items-center pt-2"
          type="button"
        >
          Stwórz grupę +
        </button>
      </a>
      <div
        class="absolute inline-block text-left right-0 mr-16"
      >
        <div
          class="flex inline"
        >
          <span
            class="mr-2"
          >
            Wyświetlane grupy: 
          </span>
          <button
            class="flex inline underline"
          >
            Wszystkie grupy
            <svg
              class=" ml-1 h-full w-[9px] items-center"
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 320 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M31.3 192h257.3c17.8 0 26.7 21.5 14.1 34.1L174.1 354.8c-7.8 7.8-20.5 7.8-28.3 0L17.2 226.1C4.6 213.5 13.5 192 31.3 192z"
              />
            </svg>
          </button>
        </div>
      </div>
    </div>
    <div
      class="relative flex flex-wrap gap-x-5 gap-y-2 bg-light_gray px-8 py-6 bg-lilly-bg rounded-md"
    >
      <div
        class="relative w-64 bg-[#59007f]  min-h-fit h-40 overflow-hidden rounded-md"
      >
        <a
          class="w-full"
          href="/group/0"
        >
          <h1
            class="flex items-center justify-center text-center pb-2 h-[50%] pt-3 text-white font-semibold text-lg"
          >
            Example Group 1
          </h1>
           
        </a>
        <p
          class="px-2 pl-2 flex items-center text-base h-[50%] text-ellipsis bg-white bottom-0 text-black"
        >
          Example desc
        </p>
      </div>
      <div
        class="relative w-64 bg-[#006400]  min-h-fit h-40 overflow-hidden rounded-md"
      >
        <a
          class="w-full"
          href="/group/2"
        >
          <h1
            class="flex items-center justify-center text-center pb-2 h-[50%] pt-3 text-white font-semibold text-lg"
          >
            Example Group 2
          </h1>
           
        </a>
        <p
          class="px-2 pl-2 flex items-center text-base h-[50%] text-ellipsis bg-white bottom-0 text-black"
        >
          Example desc 2
        </p>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
